models/articles.models.js

```diff
@@ -41,6 +41,28 @@ exports.selectCommentsByArticleId = (article_id) => {
   });
 };
 
+exports.updateArticleVotesById = (body, article_id) => {
+  return db
+    .query(
+      `UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`,
+      [body.inc_votes, article_id]
+    )
+    .then(({ rows }) => {
+      if (!rows.length) {
+        return Promise.reject({
+          status: 404,
+          msg: "Article id does not exist",
+        });
+      }
+      if (rows[0].votes < 0) {
+        return Promise.reject({
+          status: 400,
+          msg: "Cannot set votes to be a negative number",
+        });
+      }
```

Comment:
This should happen before the query, you will still be setting the votes to be a negative number through the query even though you return an error